### PR TITLE
Debian Buster image compatibility

### PR DESCRIPTION
**Pull request by amrsoll**

### CODE REVIEW

File: octoprint_mrbeam/os_health_care.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ def os_health_care(plugin):
 
 
 class OsHealthCare(object):
+    # FIXME : This should be part of the migration scripts
```
Comment (amrsoll) [inline octoprint_mrbeam/os_health_care.py]: Could be  migrated to netconnectd

Comment (amrsoll) [reply octoprint_mrbeam/os_health_care.py]: i.e. if it fails to read the network file, assume a default version

File: octoprint_mrbeam/util/device_info.py [hunk 2]
```diff
@@ -23,7 +23,6 @@ class DeviceInfo(object):
     KEY_HOSTNAME = "hostname"
     KEY_SERIAL = "serial"
     KEY_OCTOPI = "octopi"
-    KEY_IMAGE_CORRECTION_MARKERS = "image_correction_markers"
```
Comment (andywerner) [inline octoprint_mrbeam/util/device_info.py]: please keep this

File: setup.py [hunk 3]
```diff
@@ -36,7 +36,7 @@
 plugin_license = "AGPLv3"
 
 # Any additional requirements besides OctoPrint should be listed here
-plugin_requires = ["webcolors", "pillow", "lxml", "numpy==1.11.2", "picamera"]
+plugin_requires = ["webcolors", "pillow", "lxml", "numpy", "picamera"]
```
Comment (amrsoll) [inline setup.py]: Find a way to use pip 9 and update with wheels

File: octoprint_mrbeam/__init__.py [hunk 4]
```diff
@@ -210,7 +210,9 @@ def initialize(self):
         self.start_time_ntp_timer()
 
         # do os health care
-        os_health_care(self)
+        if False:
```
Comment (andywerner) [inline octoprint_mrbeam/__init__.py]: please change this.

File: octoprint_mrbeam/__init__.py [hunk 5]
```diff
@@ -2680,34 +2682,41 @@ def getHostname(self):
...
+                    # To create a custom name, use `/usr/bin/beamos_hostname XXXX`
+                    # Or consider writing to /boot/octopi-hostname
+                    self._logger.warn(
+                        "Hostname in /etc/mrbeam file ({dev_info}) does NOT match current hostname ({sys})".format(
+                            dev_info=hostname_device_info, sys=hostname_socket
+                        )
+                    )
+                    # beamos_hostname only uses 4 characters to set the hostname as MrBeam-XXXX
+                    exec_cmd(
+                        "sudo /usr/bin/beamos_hostname {}".format(
```
Comment (andywerner) [inline octoprint_mrbeam/__init__.py]: Please check if `/usr/bin/beamos_hostname` exists and provide a fallback if not

File: octoprint_mrbeam/iobeam/iobeam_handler.py [hunk 6]
```diff
@@ -75,7 +75,7 @@ class IoBeamHandler(object):
     # How to get debug info:
     #       echo "info" |  nc -U -w1 /var/run/mrbeam_iobeam.sock
 
-    SOCKET_FILE = "/var/run/mrbeam_iobeam.sock"
+    SOCKET_FILE = "/tmp/mrbeam_iobeam.sock"
```
Comment (andywerner) [inline octoprint_mrbeam/iobeam/iobeam_handler.py]: make this backwards compatible

File: octoprint_mrbeam/migrate.py [hunk 7]
```diff
@@ -817,3 +818,9 @@ def _set(path, _data, set_func, fullpath=None):
             _set(path, data, settings().setBoolean)
         settings().save()
         self._logger.info("Done.")
+
+    def track_devpi(self):
```
Comment (andywerner) [inline octoprint_mrbeam/migrate.py:830]: - needs to be called
- add try/catch
- add log output indication that this is about to happen

**Review by andywerner — CHANGES_REQUESTED** [review]

**Comment by amrsoll** [comment]
Current status: Parking this branch as we are waiting for an other beta release.

**Comment by amrsoll** [comment]
Cosing in favor of #1265

**Pull request closed without merging**